containers/ecr-viewer/integration/core_database_repo.test.ts

```diff
@@ -70,68 +81,137 @@ describe("database_repo", () => {
-      await database_repo.findEcrConditionById("12345");
+      const actual = await database_repo.findEcrConditionById("12345");
+      expect(actual).toEqual(expected);
     });
 
     it("should find all conditions named Dark Magic", async () => {
-      await database_repo.findEcrCondition({ condition: "Dark Magic" });
+      const actual = await database_repo.findEcrCondition({
+        condition: "Dark Magic",
+      });
+      expect(actual[0]).toEqual(expected);
     });
 
     it("should update the condition with a given id", async () => {
       await database_repo.updateEcrCondition("12345", {
         condition: "Extra Dark Magic",
       });
+      const actual = await database_repo.findEcrConditionById("12345");
+      expect(actual?.condition).toEqual("Extra Dark Magic");
     });
 
     it("should create a condition", async () => {
       await database_repo.createEcrCondition({
-        eICR_ID: "12345",
-        uuid: "setid",
+        eICR_ID: "54321",
+        uuid: "54321",
+        condition: "Dark Magic",
+      });
+      const actual = await database_repo.findEcrConditionById("54321");
+      expect(actual).toEqual({
+        eICR_ID: "54321",
+        uuid: "54321",
         condition: "Dark Magic",
       });
     });
 
     it("should delete a condition with a given id", async () => {
       await database_repo.deleteEcrCondition("12345");
+      const actual = await database_repo.findEcrConditionById("12345");
+      expect(actual).toBeUndefined();
     });
   });
 
   // ecr_rr_rule_summaries
   describe("ecr_rr_rule_summaries", () => {
+    const expected = {
+      uuid: "12345",
+      ecr_rr_conditions_id: "12345",
+      rule_summary: "Dark Magic",
+    };
+
+    beforeEach(async () => {
+      await database_repo.createEcrRule(expected);
+    });
+
+    afterEach(async () => {
+      await clearCore();
+    });
     it("should find a rule summary with a given uuid", async () => {
-      await database_repo.findEcrRuleById("12345");
+      const actual = await database_repo.findEcrRuleById("12345");
+      expect(actual).toEqual(expected);
     });
 
     it("should find all rule summaries named Dark Magic", async () => {
-      await database_repo.findEcrRule({ rule_summary: "Dark Magic" });
+      const actual = await database_repo.findEcrRule({
+        rule_summary: "Dark Magic",
+      });
+      expect(actual[0]).toEqual(expected);
     });
 
     it("should update the rule summary with a given id", async () => {
       await database_repo.updateEcrRule("12345", {
         rule_summary: "Extra Dark Magic",
       });
+      const actual = await database_repo.findEcrRuleById("12345");
+      expect(actual?.rule_summary).toEqual("Extra Dark Magic");
     });
 
     it("should create a rule summary", async () => {
       await database_repo.createEcrRule({
-        ecr_rr_conditions_id: "12345",
-        uuid: "setid",
+        uuid: "54321",
+        ecr_rr_conditions_id: "54321",
```

Comment:
can we use different ids here to make sure the `ById` is filtering on the correct id?